modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaAddFDSTableSchemaFieldCheck.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.parser.JavaClass;
+import com.liferay.source.formatter.parser.JavaClassParser;
+import com.liferay.source.formatter.parser.JavaMethod;
+import com.liferay.source.formatter.parser.JavaTerm;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Michael Cavalcanti
+ */
+public class UpgradeJavaAddFDSTableSchemaFieldCheck extends BaseFileCheck {
+
+	@Override
+	protected String doProcess(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		if (!fileName.endsWith(".java")) {
+			return content;
+		}
+
+		JavaClass javaClass = JavaClassParser.parseJavaClass(fileName, content);
+
+		List<String> importNames = javaClass.getImportNames();
+
+		if (!importNames.contains(
+				"com.liferay.frontend.data.set.view.table." +
+					"FDSTableSchemaBuilder")) {
+
+			return content;
+		}
+
+		return _replaceAddFDSTableSchemaField(content, javaClass);
+	}
+
+	private boolean _isFDSTableSchemaBuilderMethodCall(
+		String content, String fileContent, String methodCall) {
+
+		String variableTypeName = getVariableTypeName(
+			content, fileContent,
+			methodCall.substring(0, methodCall.indexOf(StringPool.PERIOD)),
+			true);
+
+		return variableTypeName.contains("FDSTableSchemaBuilder");
+	}
```

Comment:
You can remove this method and use `BaseSourceCheck.hasClassOrVariableName()` instead.

Reply:
@kevhlee. I did not use this method already created because I realized that it receives only one content and replicates this content when calling the method getVariableTypeName and this limits my checking, because I will be limited to check only if the variable exists in my local scope (inside my method) or if the variable exists inside my global scope, if I search the full class before checking my specific method I will open a gap for a variable with the same name that was created before to confuse my check.
I believe that when I want to check the call of a variable, first I try to look inside my local scope, i.e. inside my javamethod, if I don't find it, then I check if it is in the global scope, and then I need the entire contents of the class, in my filecontent, to check if the variable was created in the global scope. If you notice, the getVariableTypeName method does exactly the behavior I described above. I talked to Tamyris to update this method because with this correction, I could use it in my checks too. What do you think?

Reply:
@kevhlee, I changed the implementation of the method and adjusted its previous implementation according to what we talked about @natocesarrego.

Reply:
@michaelmdelcavalcanti, I'm still confused why `BaseSourceCheck.hasClassOrVariableName()` would not work here? Could you give me an example?

Reply:
I think I understand what you mean.